apps/issuance/src/issuance.service.ts

```diff
@@ -425,259 +426,305 @@ export class IssuanceService {
-          const platformConfigData = await this.issuanceRepository.getPlatformConfigDetails();
+    // Flatten the results array
+    const flattenedResults = [].concat(...results);
 
-          if (!platformConfigData) {
-            errors.push(new NotFoundException(ResponseMessages.issuance.error.platformConfigNotFound));
-            return false;
-          }
+    // Check if all emails were successfully sent
+    const allSuccessful = flattenedResults.every((result) => true === result);
 
-          this.emailData.emailFrom = platformConfigData.emailFrom;
-          this.emailData.emailTo = emailId;
-          this.emailData.emailSubject = `${process.env.PLATFORM_NAME} Platform: Issuance of Your Credential`;
-          this.emailData.emailHtml = this.outOfBandIssuance.outOfBandIssuance(emailId, organizationDetails.name, agentEndPoint);
-          this.emailData.emailAttachments = [
-            {
-              filename: 'qrcode.png',
-              content: outOfBandIssuanceQrCode.split(';base64,')[1],
-              contentType: 'image/png',
-              disposition: 'attachment'
-            }
-          ];
+    if (0 < errors.length) {
+      throw errors;
+    }
 
-          const isEmailSent = await sendEmail(this.emailData);
-          this.logger.log(`isEmailSent ::: ${JSON.stringify(isEmailSent)}`);
+    return allSuccessful;
+  } catch (error) {
+    this.logger.error(
+      `[outOfBoundCredentialOffer] - error in create out-of-band credentials: ${JSON.stringify(error)}`
+    );
+    if (0 < error?.length) {
+      const errorStack = error?.map((item) => {
+        const { message, statusCode, error } = item?.error || item?.response || {};
+        return {
+          message,
+          statusCode,
+          error
+        };
+      });
+      throw new RpcException({
+        error: errorStack,
+        statusCode: error?.status?.code,
+        message: ResponseMessages.issuance.error.unableToCreateOOBOffer
+      });
+    } else {
+      throw new RpcException(error.response ? error.response : error);
+    }
+  }
+}
 
-          if (!isEmailSent) {
-            errors.push(new InternalServerErrorException(ResponseMessages.issuance.error.emailSend));
-            return false;
+async sendEmailForCredentialOffer(sendEmailCredentialOffer: SendEmailCredentialOffer): Promise<boolean> {
+  const {
+    iterator,
+    emailId,
+    index,
+    credentialType,
+    protocolVersion,
+    attributes,
+    credentialDefinitionId,
+    outOfBandCredential,
+    comment,
+    organisation,
+    errors,
+    url,
+    apiKey,
+    organizationDetails
+  } = sendEmailCredentialOffer;
+
+  const iterationNo = index + 1;
+  try {
+    let outOfBandIssuancePayload;
+    if (IssueCredentialType.INDY === credentialType) {
+    
+      outOfBandIssuancePayload = {
+        protocolVersion: protocolVersion || 'v1',
+        credentialFormats: {
+          indy: {
+            // eslint-disable-next-line @typescript-eslint/no-unused-vars
```

Comment:
Any specific reason to disable eslint?

Reply:
We need to eliminate the isRequired attribute from the payload to avoid mapping unnecessary attributes as required. Therefore, we should create a separate variable for isRequired and ensure that it is not utilized within this function.